src/components/TodosForm/TodosForm.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Dispatch,
+  SetStateAction,
+  ChangeEvent,
+  FormEvent,
+  useState,
+  useCallback,
+} from 'react';
+
+import users from '../../api/users';
+import { getNextTodoId } from '../../utils/todo/getNextTodoId';
+import { ITodo } from '../../types/todo';
+
+interface IProps {
+  todos: ITodo[];
+  setTodos: Dispatch<SetStateAction<ITodo[]>>;
+}
+
+const TodosForm = ({ todos, setTodos }: IProps) => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleReset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const checkTodoState = () => {
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title || !userId) {
+      return checkTodoState();
+    }
+
+    const todo = {
+      id: getNextTodoId(todos),
+      title,
+      userId,
+      completed: false,
+    };
+
+    setTodos((prevState: ITodo[]) => [...prevState, todo]);
+    handleReset();
+  };
+
+  const handleChangeTitle = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setTitle(e.target.value);
+      if (titleError) {
+        setTitleError(false);
+      }
+    },
+    [titleError],
+  );
+
+  const handleChangeUserId = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setUserId(+e.target.value);
+      if (userError) {
+        setUserError(false);
+      }
+    },
+    [userError],
+  );
+
+  return (
+    <form action="/api/todos.ts" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <input
+          type="text"
+          data-cy="titleInput"
+          name="title"
+          value={title !== null ? title : ''}
```

Comment:
```suggestion
          value={title}
```
You don't need it now